### PR TITLE
Fix email validation and implement events searching.

### PR DESCRIPTION
**Pull request by Rodgers-M**
**What does this PR do?**

implements events searching and email validation

**Description of tasks to be completed?**

Validate email during registration. users should have unique emails.

**How should this be tested manually?**

Create a few events in the app and search for events based on the two parameters.

Users should be able to search events based on category or location.

Replace the search field on landing page with a get started button.

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -184,18 +184,22 @@ def resetpass():
 		return redirect(url_for('resetpass'))
 	return render_template('resetpass.html')
 
-@app.route('/searchevents')
+@app.route('/searchevents', methods=['POST'])
 def searchevents():
```
Comment (chrischiedo) [inline app/views.py]: the function/method name should be 'search_events' and not 'searchevents'.

File: app/views.py [hunk 2]
```diff
@@ -184,18 +184,22 @@ def resetpass():
...
 	"""A route to search events depending on the events category or location"""
-	parameter = request.args.get('parameter', None)
-	if parameter == None:
-		return jsonify(status="no event")
+	parameter = request.form['search']
+	if len(parameter.strip()) == 0:
```
Comment (meshnesh) [inline app/views.py]: * bad practice to use Len(SEQUENCE) as condition value, use use pylint to follow best practices.

**Comment by meshnesh** [comment]
* Did not see the implementation of email validation
* Good Implementation of events searching in the code

**Comment by betsybeth** [comment]
I cannot see any email validation according to the PR title.

**Comment by Rodgers-M** [comment]
@betsybeth  @meshnesh the email validation is now updated, in `user.py`

**Pull request merged**